src/main/java/Deadline.java

```diff
@@ -0,0 +1,14 @@
+public class Deadline extends Task {
+
+    private String by;
+
+    public Deadline (String description, String by) {
+        super(description);
+        this.by = by;
+    }
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " " + "(by: " + by + ")"; //superclass is task
```

Comment:
I would recommend to have the label 'D' as a static constant outside the method.